ansible/roles/packages/tasks/repo-redhat.yaml

```diff
@@ -1,4 +1,49 @@
 ---
+
+- name: install epel-repository
+  when:
+    - not offline_mode_enabled
+  block:
+    - name: Remove appstream repository
+      yum_repository:
+        name: appstream
+        state: absent
+      ignore_errors: True
+      when:
+        - ansible_distribution_major_version == '8'
+
+    # The AppStream repo for Centos 8 is not available from centos mirror list.
+    # AlmaLinux is 1:1 binary compatible with RHEL and subscription free.
+    # Alma linux's extra repository symlinks epel repository rpms. https://wiki.almalinux.org/repos/AlmaLinux.html
+    # Why not RHEL repo?: RHEL is RHEL though the free subscription expires once a year.
+    # it cannot be renewed until it does expire and then you have to re-register your systems to pick up the new subscription.
+    # Also RHEL repos are missing some python3 rpms that we need.
+    - name: Add Alma linux extra repository
+      yum_repository:
+        name: alma-linux-extra
+        description: Extra repository for alma linux
+        file: almalinux
+        baseurl: https://repo.almalinux.org/almalinux/$releasever/extras/$basearch/os/
+        mirrorlist: https://mirrors.almalinux.org/mirrorlist/$releasever/extras
+        enabled: yes
+        gpgkey: https://repo.almalinux.org/almalinux/RPM-GPG-KEY-AlmaLinux
+        gpgcheck: yes
+      when:
+        - ansible_distribution_major_version == '8'
+
+    # The correct epel-release rpms will be downloaded for Centos, RHEL and oracle 7 repos from in-built extra repos
+    # epel-release for Centos/RHEL/Oracle 8 will be installed using Alma linux "extra" repository
+    - name: install epel-release
+      yum:
+        name: epel-release
+        state: present
```

Comment:
Where are the in-built extra repos coming from? Just a little confused Oracle7 seems to be failing on this for some reason

Reply:
in-built extra repos are installed with the OS in `/etc/yum.repos.d/` . each distribution has its own `base`, `extra` and `appstream` repos.  Those are not always in sync 100% with their `upstream` counter part.

Reply:
added back centos 7 repositories. The changes in this PR only affects centos 8 based repos.